pkg/generators/forwarding/fluentd/generators.go

```diff
@@ -44,30 +44,67 @@ func NewConfigGenerator(includeLegacyForwardConfig, includeLegacySyslogConfig, u
 func (engine *ConfigGenerator) Generate(forwarder *logging.ClusterLogForwarderSpec) (string, error) {
 	logger.DebugObject("Generating fluent.conf using %s", forwarder)
 	//sanitize here
-	var sourceInputLabels []string
-	var sourceToPipelineLabels []string
-	var pipelineToOutputLabels []string
-	var outputLabels []string
-	var err error
+	var (
+		inputs                 sets.String
+		namespaces             sets.String
+		routeMap               logging.RouteMap
+		sourceInputLabels      []string
+		sourceToPipelineLabels []string
+		pipelineToOutputLabels []string
+		outputLabels           []string
+		err                    error
+	)
 
-	inputs, namespaces := gatherSources(forwarder)
+	// Provide inputs and inputsPipelines for legacy forwarding protocols
+	// w/o a user-provided ClusterLogFowarder instance to enable inputs and
+	// inputs-to-pipelines a.k.a. route map template generation.
+	if engine.includeLegacyForwardConfig || engine.includeLegacySyslogConfig {
+		inputs = sets.NewString(
```

Comment:
If legacy LF is used, it must collect all inputs?

Reply:
For legacy LF there was no config option to declare which logs to collect or not. Users could only configure which logs they accepted on fluentd/syslog and drop all others.